### PR TITLE
chore(netflix): convert reservation report module

### PR DESCRIPTION
**Pull request by icfantv**
* convert reservation report module to TS
* add missing tests for business logic in service.

### CODE REVIEW

**Comment by icfantv** [comment]
@anotherchrisberry @jrsquared @christopherthielen PTAL

File: app/scripts/modules/netflix/report/reservationReport.component.spec.ts [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
+import {RESERVATION_REPORT_COMPONENT} from './reservationReport.component';
+import {ReservationReportReader} from 'netflix/report/reservationReport.read.service';
+import {AccountService} from 'core/account/account.service';
 
 describe('Directives: reservation report', function () {
 
-  require('./reservationReport.directive.html');
-
-  beforeEach(window.module(require('./reservationReport.directive')));
+  require('./reservationReport.component.html');
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.component.spec.ts]: import

File: app/scripts/modules/netflix/report/reservationReport.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+  beforeEach(mock.module(RESERVATION_REPORT_COMPONENT));
   beforeEach(function() {
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.component.spec.ts]: arrow function

File: app/scripts/modules/netflix/report/reservationReport.component.spec.ts [hunk 3]
```diff
@@ -1,26 +1,38 @@
...
     jasmine.addMatchers(matchers);
   });
 
-  beforeEach(window.inject(function ($rootScope, $compile, reservationReportReader, $httpBackend, accountService, $q) {
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.component.spec.ts]: arrow function

File: app/scripts/modules/netflix/report/reservationReport.component.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
+import {IComponentController, IComponentOptions, ILogService, IScope, module} from 'angular';
+
+import {
+  IExtractedReservation, IReservationReport, RESERVATION_REPORT_READER,
+  ReservationReportReader
+} from './reservationReport.read.service';
+
+require('./reservationReport.component.less');
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.component.ts]: import

File: app/scripts/modules/netflix/report/reservationReport.component.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        this.viewState.loading = false;
+        this.viewState.error = true;
+      });
+
+    this.$scope.$watchCollection(() => [this.instanceType, this.account, this.region, this.isVpc, this.zones], () => this.setReportData());
+  }
+}
+
+class ReservationReportComponent implements IComponentOptions {
+  public bindings: any = {
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.component.ts]: Do any of these need to be two-way bindings? I think they can all be one way.

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  }));
+
+  describe('test reservation retrieval', () => {
+
+    beforeEach(() => {
+      $http.whenGET(`${SETTINGS.gateUrl}/reports/reservation/v2`).respond(200, []);
+    });
+
+    it('should return a promise', () => {
+
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: whitespace

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+      return reservation;
+    }
+
+    it('should return an empty array if passed an empty array', () => {
+      expect(reader.extractReservations([], 'account', 'region', 'instanceType').length).toBe(0);
+    });
+
+    it('should only extract the reservations whose regions, instanceTypes, and account match', () => {
+
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: whitespace

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      const accounts: string[] = ['test', 'prod']; // 0, 1
+      const regions: string[] = ['us-west-1', 'us-west-2', 'us-east-1', 'eu-west-1']; // 0, 1, 2, 3
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: why the counting comments?

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+          instanceTypes.forEach((instanceType: string, k: number) => {
+            reservations.push(getReservation(account, region, instanceType, `${i}${j}${k}`));
+          });
+        });
+      });
+
+      const account = 'test';
+      const region = 'us-west-2';
+      const instanceType = 'small';
+      const extracted: IExtractedReservation[] =
```
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: Have we started enforcing an 80 character line limit? Some of these line wrappings seem overzealous.

**Comment by jrsquared** [comment]
Should be `chore(...` not `core(...`

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts (hunk 8 above)
Comment (icfantv) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: I put them in there so the ID could be easily discerned.  If you no like, I'll take them out.  I tried to think of a better way to generate a bunch of data without having gobs of custom and repeated stuff.  Totally open to better suggestions here.

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts (hunk 9 above)
Comment (icfantv) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: In my new, fatherly role, I've endorsed readability.  But seriously, if you don't think it makes it more readable, I'll remove it.  I thought it did.

File: app/scripts/modules/netflix/report/reservationReport.component.ts (hunk 5 above)
Comment (icfantv) [inline app/scripts/modules/netflix/report/reservationReport.component.ts]: Totes.  Good eye there.

File: app/scripts/modules/netflix/report/reservationReport.component.spec.ts (hunk 3 above)
Comment (icfantv) [inline app/scripts/modules/netflix/report/reservationReport.component.spec.ts]: I think this was an errant comment?  Only function I can see here is for the `addMatchers`.

File: app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts (hunk 9 above)
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.read.service.spec.ts]: More readable if you have an 800x600 display, maybe.

File: app/scripts/modules/netflix/report/reservationReport.component.spec.ts (hunk 3 above)
Comment (jrsquared) [inline app/scripts/modules/netflix/report/reservationReport.component.spec.ts]: `window.inject(function` ?

**Pull request merged**